### ISSUE TITLE
The extension is slowing down my vscode

### ISSUE THREAD

**Issue by gilmishal** [imliam/vscode-inline-parameters#13]
As you can see in the image below, this is me profiling over a short period of time, after experiencing my vscode being extremely slow. I was profiling for a few minutes, and your extension is the one taking most of the time. 

I want to say that I very much appreciate your work and I hope to use it all the time, but unfortunately if it carries on being slow I would have to stop using it, and I don't realy want to.
For the record, I was developing with Angular and TypeScript.

![image](https://user-images.githubusercontent.com/9153443/84399892-0b73a480-ac0a-11ea-8997-562500d55d65.png)

**Comment by imliam** [comment]
Yeah - unfortunately this extension can be a little bit slow because it currently does a lot of work each pass - parsing the entire current file and figuring out what all the parameters are, essentially each time you type. There's some debouncing to stop it from happening all the time, but could still be improved.

I have some ideas on how to speed it up and prevent this from having to happen so frequently when the extension is enabled - I'll see what I can come up with.

**Comment by MrMage** [comment]
@gilmishal sorry for hijacking this thread, but would you mind letting me know how you profiled your extensions? I only found [this](https://www.reddit.com/r/vscode/comments/7eijox/how_do_you_profile_vs_code_performance/dq6j73c/), but it doesn't look like it breaks down time spent per extension.
